packages/emotion/src/withStyle.tsx

```diff
@@ -111,7 +112,7 @@ const withStyle = decorator(
       }
       const themeOverride = getComponentThemeOverride(
         theme,
-        displayName,
+        [displayName, ComposedComponent.componentId],
```

Comment:
At this point can't we jus set `displayName` to be `ComposedComponent.componentId`?
Whats the point of keeping the `.displayName` and `.name` if we are not even sure they will contain any legit info about the component?

If we only use the `.componentId` then we don't have to do these kind of ugly checks in the `getComponentThemeOverride` function.

Reply:
I think it should be setting the displayName to `ComposedComponent.componentId || ComposedComponent.displayName`, just in case it is missing from somewhere, but primarily using the componentId, yes.

Also, the static is missing at: Calendar.Day, InPlaceEdit, InlineListItem, InlineSVG, SVGIcon (all components with @makeStyle should have it, maybe even the others too, in case we want to use the componentId for something else too.)